### PR TITLE
Check whether server has new templates when updating SinopiaResourceTemplates

### LINKED ISSUE #524: newly uploaded Resource Templates do not show up in Resource Template list in other sessions
on DEV
in Safari
I successfully uploaded 2 new ResourceTemplates (I see the blue highlighted message telling me they were uploaded, and they appear in the list "Available Resource Templates in Sinopia".

I went to Firefox, where I was already logged in, and those 2 new templates are not in the list "Available Resource Templates in Sinopia". I refreshed the browser window and they are still not there. I logged out and back in and they are still not there. I quit Firefox, restarted, re-logged in, and now I can see them.

### PR DESCRIPTION
**Pull request by mjgiarlo**
Fixes #524

This PR adds a new condition to `SinopiaResourceTemplates.componentDidUpdate()` that issues a HEAD request to the LD4P group to get its ETag, and compares it against the ETag stored in component state. If the ETag has changed, it means the LD4P group has changed which in practice is a signal that one or more resource templates have been added to the group since the component loaded. This makes sure that server conditions can defeat local caching that is currently preventing sessions from seeing templates created in other sessions (without using a hard refresh).

@michelleif has confirmed in #524 that the changes in this PR fix the bug (tested in dev).

Also includes:
* Send `accept` header when getting a resource template: use this instead of `acceptEncoding`, which does not specify mime types (but, rather, things like gzip encoding). This is unrelated to the PR and was patched in opportunistically to prevent bugs and misunderstandings.
* Expand eslint coverage and reduce number of warnings: deal with a bunch of new eslint errors and warnings, and allow `console.error` to surface real errors.

### CODE REVIEW

File: __tests__/components/App.test.js [hunk 1]
```diff
@@ -56,17 +56,20 @@ describe("#routes", () => {
...
     // TODO: needs fixing for re-done authN approach, see #528
     it.skip('/templates renders the Login component (with location props) if the user is not authenticated', () => {
-      saveState({loginJwt: {}, isAuthenticated: false}, 'jwtAuth')
-      const component = renderRoutes("/templates")
-      expect(component.find(Login).length).toEqual(1)
-      expect(component.find(Login).prop('location')).toBeDefined()
+      // TODO: This method is undefined
+      // saveState({loginJwt: {}, isAuthenticated: false}, 'jwtAuth')
+      // const component = renderRoutes("/templates")
+      // TODO: Login is not defined or imported
```
Comment (jmartin-sul) [inline __tests__/components/App.test.js]: yeah, that component got removed in #460 (and the test needed reworking anyway, so skipped and punted w/ the intention of revisiting in #539).

File: src/components/editor/SinopiaResourceTemplates.jsx [hunk 2]
```diff
@@ -18,14 +18,34 @@ class SinopiaResourceTemplates extends Component {
...
+      // Reset errors when component updates, else errors from a prior update will prevent RTs from displaying
+      this.setState({ errors: [] })
+      await this.fetchResourceTemplatesFromGroups()
+    }
+  }
 
-    await this.fetchResourceTemplatesFromGroups()
+  serverHasNewTemplates = async () => {
+    try {
+      const response = await fetch(`${Config.sinopiaServerBase}/repository/${Config.defaultSinopiaGroupId}`, {
```
[the remark this answers is not in the document]
Comment (mjgiarlo) [reply src/components/editor/SinopiaResourceTemplates.jsx]: Ah! I missed that entirely. I am going to push this in my next branch update.

File: __tests__/components/editor/SinopiaResourceTemplates.test.js [hunk 3]
```diff
@@ -30,26 +30,95 @@ describe('<SinopiaResourceTemplates />', () => {
...
 
       expect(fetchSpy).not.toHaveBeenCalled()
+      expect(serverCheckSpy).toHaveBeenCalledTimes(1)
+      expect(setStateSpy).not.toHaveBeenCalled()
+    })
+  })
+
+  describe('serverHasNewTemplates()', () => {
+    it('returns false when etag has not changed', async () => {
+      global.fetch = jest.fn().mockImplementation(async () => {
```
[the remark this answers is not in the document]
Comment (mjgiarlo) [reply __tests__/components/editor/SinopiaResourceTemplates.test.js]: Good call. I am dumping this fetch stuff in favor of using the sinopia client, too.

File: src/components/editor/SinopiaResourceTemplates.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (mjgiarlo) [reply src/components/editor/SinopiaResourceTemplates.jsx]: Pushed. Resolving.

File: __tests__/components/App.test.js (hunk 1 above)
Comment (mjgiarlo) [reply __tests__/components/App.test.js]: ok, np! Commented it out to make eslint happy. Resolving.

File: __tests__/components/editor/SinopiaResourceTemplates.test.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (mjgiarlo) [reply __tests__/components/editor/SinopiaResourceTemplates.test.js]: Pushed, resolving.

File: src/components/LoginPanel.jsx [hunk 4]
```diff
@@ -49,10 +49,8 @@ class LoginPanel extends Component {
 
   handleSignout = () => {
     this.props.currentUser.globalSignOut({
-      onSuccess: (result) => {
-        console.log('handleSignout result: ' + result)
+      onSuccess: () => {
```
Comment (jmartin-sul) [inline src/components/LoginPanel.jsx]: oh cool, i didn't realize the callback method could omit the param without causing issues for [a caller that supplies one](https://github.com/aws-amplify/amplify-js/blob/master/packages/amazon-cognito-identity-js/src/CognitoUser.js#L1634-L1656), but JS is even more tolerant of parameter mismatches than i remembered.

Comment (mjgiarlo) [reply src/components/LoginPanel.jsx]: Yes, very much so!

File: src/components/editor/SinopiaResourceTemplates.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (jmartin-sul) [reply src/components/editor/SinopiaResourceTemplates.jsx]: thanks!

File: __tests__/components/editor/PropertyResourceTemplate.test.js [hunk 5]
```diff
@@ -21,7 +21,7 @@ describe('<PropertyResourceTemplate />', () => {
     },
     reduxPath: ['resourceTemplate:test']
   }
-  const testId = jest.spyOn(shortid, 'generate').mockReturnValue('abcd45')
+  jest.spyOn(shortid, 'generate').mockReturnValue('abcd45')
```
Comment (jgreben) [inline __tests__/components/editor/PropertyResourceTemplate.test.js]: Do you want to use this instead, similar to`__tests__/components/editor/ResourceTemplateForm.test.js`?

```
shortid.generate = jest.fn().mockReturnValue('abcd45')
```

File: __tests__/reducers/index.test.js [hunk 6]
```diff
@@ -83,7 +82,7 @@ describe(`Takes a resource template ID and populates the global state`, () => {
   })
 
   it('handles SET_RESOURCE_TEMPLATE', () => {
-    const testId = jest.spyOn(shortid, 'generate').mockReturnValue(0)
+    jest.spyOn(shortid, 'generate').mockReturnValue(0)
```
Comment (jgreben) [inline __tests__/reducers/index.test.js]: `shortid.generate = jest.fn().mockReturnValue(0)`?

File: __tests__/reducers/index.test.js [hunk 7]
```diff
@@ -132,7 +131,7 @@ describe(`Takes a resource template ID and populates the global state`, () => {
   })
 
   it('tests with a more realistic payload with defaults', () => {
-    const testId = jest.spyOn(shortid, 'generate').mockReturnValue(0)
+    jest.spyOn(shortid, 'generate').mockReturnValue(0)
```
Comment (jgreben) [inline __tests__/reducers/index.test.js]: `shortid.generate = jest.fn().mockReturnValue(0)`?

File: __tests__/components/editor/PropertyResourceTemplate.test.js (hunk 5 above)
Comment (mjgiarlo) [reply __tests__/components/editor/PropertyResourceTemplate.test.js]: Ah, yes. Good call. Changed and pushed. Thanks! Resolving.

File: __tests__/reducers/index.test.js (hunk 6 above)
Comment (mjgiarlo) [reply __tests__/reducers/index.test.js]: as above!

File: __tests__/reducers/index.test.js (hunk 7 above)
Comment (mjgiarlo) [reply __tests__/reducers/index.test.js]: as above!

File: __tests__/components/App.test.js [hunk 8]
```diff
@@ -56,17 +56,20 @@ describe("#routes", () => {
...
       expect(component.find(ImportResourceTemplate).length).toEqual(1)
     })
 
     // TODO: needs fixing for re-done authN approach, see #528
     it.skip('/templates renders the Login component (with location props) if the user is not authenticated', () => {
-      saveState({loginJwt: {}, isAuthenticated: false}, 'jwtAuth')
-      const component = renderRoutes("/templates")
-      expect(component.find(Login).length).toEqual(1)
-      expect(component.find(Login).prop('location')).toBeDefined()
+      // TODO: This method is undefined
```
[the remark this answers is not in the document]
Comment (mjgiarlo) [reply __tests__/components/App.test.js]: Yes and yes. This is included w/ the work commented on and ticketed on line 65, which I believe @jmartin-sul was taking on soon.

File: __tests__/components/editor/SinopiaResourceTemplates.test.js [hunk 9]
```diff
@@ -60,15 +116,17 @@ describe('<SinopiaResourceTemplates />', () => {
...
-    const containsGroups = {
-      response: {
-        body: {
-          contains: [
-            'ld4p'
-          ]
-        }
-      }
-    }
+    // TODO: Restore this once RTs are stored in multiple groups
```
[the remark this answers is not in the document]
Comment (mjgiarlo) [reply __tests__/components/editor/SinopiaResourceTemplates.test.js]: @ndushay I'm going to remove the TODOs from this file.

[the remark this answers is not in the document]
Comment (mjgiarlo) [reply __tests__/components/editor/SinopiaResourceTemplates.test.js]: pushed.

**Pull request merged**